skimage/feature/tests/test_texture.py

```diff
@@ -176,6 +176,42 @@ def test_correlation(self):
         np.testing.assert_almost_equal(energy[0, 0], 0.71953255)
         np.testing.assert_almost_equal(energy[1, 0], 0.41176470)
 
+    def test_mean(self):
+        result = graycomatrix(
+            self.image, [1], [0, np.pi / 2], 4, normed=True, symmetric=True
+        )
```

Comment:
Perhaps this could be moved just outside (before) this test function, since it's used and reused by other test functions?

Reply:
:thinking: I just wanted to double-check, but locally I can only `$ spin test skimage/feature/tests/test_texture.py` (I'd like to `$ spin test skimage/feature/tests/test_texture.py::test_mean` but that's a `pytest` functionality, not a `spin test` one...). How do you run tests locally?

Reply:
I also use `spin test` to run tests.

The only way I can think of moving these things outside the test cases is doing something like this:
```suggestion
    def base_prop(self, prop_name, ref):
        result = graycomatrix(
            self.image, [1], [0, np.pi / 2], 4, normed=True, symmetric=True
        )
        prop = graycoprops(result, prop_name)[0, 0]

        # Reference value was calculated by hand and is close to original source if precision 3 is used.
        np.testing.assert_almost_equal(prop, ref)

    def test_mean(self):
        self.base_prop("mean", 1.29166667)

    def test_variance(self):
        self.base_prop('variance', 1.03993055)

    def test_std(self):
        self.base_prop('std', 1.01976985)

    def test_entropy(self):
        self.base_prop('entropy', 2.09472904)
```
I believe it would make more sense to keep them independent like the previous tests, even though it's kinda repetitive.

Reply:
Or maybe `result` can be passed as an argument to all these functions?

Reply:
How can I do that?

There ins't a test case in the test suite that does something like this.
There are parameterized tests, but that's not the case.

In my understanding of testing scripts, each class method that begins with the `test` is executed using a different context, so result would be computed each time either way, unless I add a new attribute like `image` (that is reused) in every test.

Reply:
It is also possible to use a single method to test multiple properties, but it breaks the pattern of previous tests.

Reply:
@lagru @mkcor @stefanv 
How should I proceed with this?
May I leave the tests like this?

Reply:
I will mark the conversations as resolved then.